Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -227,8 +292,52 @@ endRide handle@ServiceHandle {..} rideId req = withLogTag ("rideId-" <> rideId.g
         >>= fromMaybeM (MerchantPaymentMethodNotFound paymentMethodId.getId)
     let mbPaymentUrl = DMPM.getPostpaidPaymentUrl =<< mbPaymentMethod
     let mbPaymentMethodInfo = DMPM.mkPaymentMethodInfo <$> mbPaymentMethod
-
+    smsCfg <- asks (.smsCfg)
     withTimeAPI "endRide" "notifyCompleteToBAP" $ notifyCompleteToBAP booking updRide newFareParams mbPaymentMethodInfo mbPaymentUrl
+
+    case req of
+      CallBasedReq callBasedEndRideReq -> do
+        let requestor = callBasedEndRideReq.requestor
+        _ <- C.catchAll
```

Comment:
code much duplicated for  `CallBasedReq` and `DashboardReq`

Reply:
working on adding the function in the service handle for both start ride and endride